app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -612,96 +616,123 @@ private double getPokemonLevel(Bitmap pokemonImage){
-            if(isFemale){
+            if (isFemale) {
                 pokemonName = getResources().getString(R.string.pokemon029);
-            }else{
+            } else {
                 pokemonName = getResources().getString(R.string.pokemon032);
             }
         }
-        //SaveImage(name, "name");
-        // TODO : Check rectangle and color
+        name.recycle();
+        return pokemonName;
+    }
+
+    /**
+     * gets the candy name from a pokenon image
+     * @param pokemonImage the image of the whole screen
+     * @return the candy name, or "" if nothing was found
+     */
+    private String getCandyNameFromImg(Bitmap pokemonImage){
         Bitmap candy = Bitmap.createBitmap(pokemonImage, displayMetrics.widthPixels / 2, (int) Math.round(displayMetrics.heightPixels / 1.3724285), (int) Math.round(displayMetrics.widthPixels / 2.057), (int) Math.round(displayMetrics.heightPixels / 38.4));
         candy = replaceColors(candy, 68, 105, 108, Color.WHITE, 200);
         tesseract.setImage(candy);
+        String candyName ="";
         //System.out.println(tesseract.getUTF8Text());
         //SaveImage(candy, "candy");
         try {
             candyName = tesseract.getUTF8Text().trim().replace("-", " ").split(" ")[candyOrder].replace(" ", "").replace("1", "l").replace("0", "o");
             candyName = new StringBuilder().append(candyName.substring(0, 1)).append(candyName.substring(1).toLowerCase()).toString();
-        }catch(StringIndexOutOfBoundsException e){
-            candyName = pokemonName; //Default for not finding candy name
+        } catch (StringIndexOutOfBoundsException e) {
+            candyName = "";
         }
+        candy.recycle();
+        return candyName;
+    }
+
+    /**
+     * get the pokemon hp from a picture
+     * @param pokemonImage the image of the whole screen
+     * @return an integer of the interpreted pokemon name, 10 if scan failed
+     */
+    private int getPokemonHPFromImg(Bitmap pokemonImage){
+        int pokemonHP = 10;
         Bitmap hp = Bitmap.createBitmap(pokemonImage, (int) Math.round(displayMetrics.widthPixels / 2.8), (int) Math.round(displayMetrics.heightPixels / 1.8962963), (int) Math.round(displayMetrics.widthPixels / 3.5), (int) Math.round(displayMetrics.heightPixels / 34.13333333));
         hp = replaceColors(hp, 55, 66, 61, Color.WHITE, 200);
         tesseract.setImage(hp);
-        //System.out.println(tesseract.getUTF8Text());
         String pokemonHPStr = tesseract.getUTF8Text();
-
-        //Check if valid pokemon TODO find a better method of determining whether or not this is a pokemon image
-        if(pokemonHPStr.contains("/")) {
+        if (pokemonHPStr.contains("/")) {
             try {
                 pokemonHP = Integer.parseInt(pokemonHPStr.split("/")[1].replace("Z", "2").replace("O", "0").replace("l", "1").replaceAll("[^0-9]", ""));
             } catch (java.lang.NumberFormatException e) {
                 pokemonHP = 10;
```

Comment:
This assignment is also redundant with the initialization, and we can't drop the initialization here.